barretenberg/cpp/src/barretenberg/common/jemalloc_helper.h

```diff
@@ -0,0 +1,84 @@
+
+// Aztec-specific: We include this file from
+// https://github.com/tabokie/rocksdb/blob/54dd1848405edeaf63a96432ec11471a133a7633/port/jemalloc_helper.h
+// to help with clang issues with linking jemalloc. File replicated below.
+
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#if defined(__clang__)
+// glibc's `posix_memalign()` declaration specifies `throw()` while clang's
+// declaration does not. There is a hack in clang to make its re-declaration
+// compatible with glibc's if they are declared consecutively. That hack breaks
+// if yet another `posix_memalign()` declaration comes between glibc's and
+// clang's declarations. Include "mm_malloc.h" here ensures glibc's and clang's
+// declarations both come before "jemalloc.h"'s `posix_memalign()` declaration.
+//
+// This problem could also be avoided if "jemalloc.h"'s `posix_memalign()`
+// declaration did not specify `throw()` when built with clang.
+#include <mm_malloc.h>
+#endif
+
+#ifdef ROCKSDB_JEMALLOC
+#ifdef __FreeBSD__
+#include <malloc_np.h>
+#else
+#define JEMALLOC_MANGLE
+#include <jemalloc/jemalloc.h>
+#endif
+
+#ifndef JEMALLOC_CXX_THROW
+#define JEMALLOC_CXX_THROW
+#endif
+
+#if defined(OS_WIN) && defined(_MSC_VER)
+
+// MSVC does not have weak symbol support. As long as ROCKSDB_JEMALLOC is defined,
+// Jemalloc memory allocator is used.
+static inline bool HasJemalloc()
+{
+    return true;
+}
+
+#else
+
+// Declare non-standard jemalloc APIs as weak symbols. We can null-check these
```

Comment:
There is some tricky stuff to get it from here to working, but in theory it's just making jemalloc emit as jemalloc and not malloc and then linking the header properly